Update documentation for post-deployment scripts. (#3070)
actions/virtual-environments: help/CreateImageAndAzureResources.md

- [Go](https://github.com/actions/go-versions)
- [Node](https://github.com/actions/node-versions)
- [Boost](https://github.com/actions/boost-versions)

### Post-generation scripts

> Scripts for setting up some specific configuration after packer build procedure are placed in `post-generation` folder. These scripts are intended to apply configuration which cannot be covered during packer build procedure. Scripts are required, since packer build and default agent users are not the same and some specific configuration should be applied after build process under agent's user.

Reviewer: ```suggestion
Extra configuration should be applied after packer build process, because packer build user and default agent's user are not the same. These configuration scripts ar located in the `post-generation` folder.
```

> Detailed description for the scripts:

Reviewer 2: I think we should add info where these scripts are located on machine (they are copied to machine during image generation) and may be add simple command how to run all of them.

Author: Thank you for the comments, I'll definitely add this information shortly.

#### Ubuntu

> - **cleanup-logs.sh** - wipes all log files, which were generated during build procedure

Reviewer: ```suggestion
- **cleanup-logs.sh** - removes all build process logs from the machine
```

- **environment-variables.sh** - replaces `$HOME` with the default user's home directory for environmental variables related to the default user home directory

> - **homebrew-permissions.sh** - resets brew repository directory to make the brew clean after rights change for /home directory

Reviewer: ```suggestion
- **homebrew-permissions.sh** - changes brew repository folder owner to make the brew clean after changing permissions for /home directory
```

> - **rust-permissions.sh** - fixes permissions for the Rust folder. Detailed issue explanation is avaliable by [the link](https://github.com/actions/virtual-environments/issues/572).

Reviewer: ```suggestion
- **rust-permissions.sh** - fixes permissions for the Rust folder. Detailed issue explanation is provided in [virtual-environments/issues/572](https://github.com/actions/virtual-environments/issues/572).
```

#### Windows
- **Choco.ps1** - contains dummy command to cleanup orphaned packages to avoid initial delay for future choco commands

> - **Dotnet.ps1** - adds `$env:USERPROFILE\.dotnet\tools` direcort to the PATH

Reviewer: ```suggestion
- **Dotnet.ps1** - adds `$env:USERPROFILE\.dotnet\tools` directory to the PATH
```

- [Go](https://github.com/actions/go-versions)
- [Node](https://github.com/actions/node-versions)
- [Boost](https://github.com/actions/boost-versions)

### Post-generation scripts
Extra configuration should be applied after packer build process, because packer build user and default agent's user are not the same. These configuration scripts are located in the `post-generation` folder. 

> The scripts are copied to the virtual machines during the packer build and can be found in the `/opt/post-generation` directory for Ubuntu and `C:\post-generation` directory for Windows images.

Reviewer: ```suggestion
The scripts are copied to the virtual machines during the packer build to `/opt/post-generation` directory for Ubuntu and to `C:\post-generation` directory for Windows images.
```

> Use the following commands to run these scripts::

Reviewer 2: Let's convert this line to header

> Use the following commands to run these scripts::

Reviewer 2: ```suggestion
#### Running scripts
```

> In order to run all post-generation scripts it is possible to use the following commands:

Reviewer: ```suggestion
Use the following commands to run these scripts:
```

- **Ubuntu**

>         for i in $(find /opt/post-generation -mindepth 1 -maxdepth 1 -type f -name '*.sh'); do bash "$script"; done

Reviewer 2: You can use find --exec instead of for loop.
https://www.baeldung.com/linux/find-exec-command

- **Windows**

>         Get-ChildItem C:\post-generation -Filter *.ps1 | ForEach-Object { $_.FullName }

Reviewer 2: Will this command invoke scripts?
I think we need to do something like `Get-ChildItem "C:\post-generation" -Filter *.ps1 | ForEach-Object { & $_.FullName }`?

> Scripts detailed description:

Reviewer 2: ```suggestion
#### Script details
```

> Scripts detailed description:

Reviewer 2: Let's convert this line to header

> Detailed description for the scripts:

Reviewer: ```suggestion
Scripts detailed description:
```

- [Go](https://github.com/actions/go-versions)
- [Node](https://github.com/actions/node-versions)
- [Boost](https://github.com/actions/boost-versions)

### Post-generation scripts

> Extra configuration should be applied after packer build process, because packer build user and default agent's user are not the same. These configuration scripts are located in the `post-generation` folder. 

Reviewer 3: ```suggestion
### Post-generation scripts
The user, created during the image generation, does not exist in the result VHD hence some configuration files related to the user's home directory need to be changed as well as the file permissions for some directories. Scripts for that are located in the `post-generation` folder in the repository:
- Windows https://github.com/actions/virtual-environments/tree/main/images/win/post-generation
- Linux https://github.com/actions/virtual-environments/tree/main/images/linux/post-generation
```

> The scripts are copied to the virtual machines during the packer build to `/opt/post-generation` directory for Ubuntu and to `C:\post-generation` directory for Windows images.

Reviewer 3: ```suggestion
The scripts are copied to the VHD during the image generation process to the following paths:
- Windows `C:\post-generation`
- Linux  `/opt/post-generation` 
▲

```

The user, created during the image generation, does not exist in the result VHD hence some configuration files related to the user's home directory need to be changed as well as the file permissions for some directories. Scripts for that are located in the `post-generation` folder in the repository:
- Windows https://github.com/actions/virtual-environments/tree/main/images/win/post-generation
- Linux https://github.com/actions/virtual-environments/tree/main/images/linux/post-generation

The scripts are copied to the virtual machines during the packer build to `/opt/post-generation` directory for Ubuntu and to `C:\post-generation` directory for Windows images.

#### Running scripts

##### Ubuntu

        find /opt/post-generation -mindepth 1 -maxdepth 1 -type f -name '*.sh' -exec bash {} \;

##### Windows

        Get-ChildItem C:\post-generation -Filter *.ps1 | ForEach-Object { & $_.FullName }

#### Script details

##### Ubuntu

- **cleanup-logs.sh** - removes all build process logs from the machine
- **environment-variables.sh** - replaces `$HOME` with the default user's home directory for environmental variables related to the default user home directory

> - **homebrew-permissions.sh** - changes brew repository folder owner to make the brew clean after changing permissions for `/home` directory

Reviewer 3: ```suggestion
- **homebrew-permissions.sh** - Resets homebrew repository directory by running `git reset --hard` to make the working tree clean after chmoding /home and changes the repository directory owner to the current user
```

- **rust-permissions.sh** - fixes permissions for the Rust folder. Detailed issue explanation is provided in [virtual-environments/issues/572](https://github.com/actions/virtual-environments/issues/572).

##### Windows

- **Choco.ps1** - contains dummy command to cleanup orphaned packages to avoid initial delay for future choco commands
- **Dotnet.ps1** - adds `$env:USERPROFILE\.dotnet\tools` directory to the PATH
- **InternetExplorerConfiguration** - turns off the Internet Explorer Enhanced Security feature

> - **Msys2FirstLaunch.ps1** - creates user profile at the first launch

Reviewer 3: user profile for what?
